Make focus widget number go UP when "optimizing" learning

RenechCDDA opened the pull request:
#### Summary
Interface "Make focus widget number go UP when 'optimizing' learning"

#### Purpose of change
Alternative to #84059

#### Describe the solution
Once again we abuse human psychology

#### Describe alternatives you've considered
Honestly I have been taken by arguments for removing focus entirely. But that is a ton of work, so maybe this little bit will be enough.

#### Testing
<img width="90" height="29" alt="image" src="https://github.com/user-attachments/assets/19a993d7-ee38-48c1-8808-447ad5363792" />

"Focus" continues to rise as you're spending the most possible focus_pool

"Focus" drops when you're not spending as much as possible.


#### Additional context

RenechCDDA committed:
Make focus widget number go UP when "optimizing" learning

PatrikLundell commented:
So you're lying to the player, by telling him that the slower learning gets, the better it is, until the learning gets so slow it's down to a crawl, at which point it's finally indicated as good old neutral, instead of that horrible bad)? Grind as much as possible, because that's good for you. Don't dare take breaks from grinding, as you'll soon go down into red.

This is even worse than hiding the value: that didn't actively lie, at least.

RenechCDDA committed:
Updated tests for changes to focus widget

Most of them were simply checking spacing. In those cases we were using the exact same number of characters so there was no need for any significant examination.

The specific check for the focus widget working as intended got some comments, to explain what was happening.

I-am-Erk commented:
I missed this before commenting on the other PR sorry. That's what I get for taking so long to review the things I flagged for myself. Sigh.

I think this is a way better step than the other one, but I'd have two suggestions:
- there should be some indication when you are overspending focus. It has been too long since I reviewed how our current focus system works, since it is IMO broken and needs rewriting anyway, but I believe it is quite important to let a player know they can't just wander off afk doing an activity that is going to deplete their focus pool very quickly.
- it's probably better still to represent this descriptively than with numbers. MilosRasic put it perfectly: this is something you can manage, but not micromanage.

Tiareth commented:
If you decide to merge this, before that. please test how it works with supernatural mods. How new focus values adress and signal your ability to cast. For example, Magiclysm is full of spells whose success chance heavily relies on focus.
